### PR TITLE
HOA-86 feat(validation-messages): add support for controlName input

### PR DESCRIPTION
**Pull request by dawidkostka-va**
Add support for controlName input

### CODE REVIEW

File: apps/validation-messages-example-app/src/app/app.component.html [hunk 1]
```diff
@@ -1,28 +1,51 @@
...
+<form [formGroup]="form" class="form">
+  <div>
+    <mat-form-field>
+      <mat-label>Email</mat-label>
+      <input matInput type="email" formControlName="email" />
+      <mat-error>
+        <!-- component retrieves form control from mat-form-field -->
+        <!-- global messages are extended/overridden by a local config -->
+        <ng-validation-messages
+          [validationMessages]="{
```
Comment (mateuszbasinski) [inline apps/validation-messages-example-app/src/app/app.component.html:10]: I think input should be called errorsMessages

File: projects/validation-messages/README.md [hunk 2]
```diff
@@ -1,39 +1,48 @@
...
 
 # Usage
 
 ### 1. Import ValidationMessagesModule
 
-Import ValidationMessagesModule in the module of your application you'd like to use ng-validation-messages component.
+In the module where you want to use the ng-validation-messages component, import the ValidationMessagesModule as follows:
 
-```
+```ts
```
Comment (mateuszbasinski) [inline projects/validation-messages/README.md:15]: ts?

File: apps/validation-messages-example-app/src/app/app.component.html (hunk 1 above)
Comment (dawidkostka-va) [reply apps/validation-messages-example-app/src/app/app.component.html]: I've changed its name to errorMessages

File: projects/validation-messages/README.md [hunk 3]
```diff
@@ -42,66 +51,125 @@ export class ValidationMessagesConfigService {
...
+export interface ValidationMessage {
+  message: string;
+  validatorValue?: string;
+  pattern?: string;
+  validatorValueParser?: (value: any) => string;
+  templateMatcher?: RegExp;
+}
+```
+`validatorValue`: specifies the name of the property under the validator name in Form Control errrors object from where the value for interpolation will be taken.
```
Comment (mateuszbasinski) [inline projects/validation-messages/README.md:169]:
```suggestion
`validatorValue`: specifies the name of the property under the validator name in Form Control errors object from where the value for interpolation will be taken.
```

File: projects/validation-messages/src/lib/components/validation-messages/validation-messages.component.ts [hunk 4]
```diff
@@ -1,36 +1,55 @@
...
   parsedApiErrorMessages: string[] = [];
   valueChanges: Subscription | null = null;
   private ngUnsubscribe: Subject<void> = new Subject<void>();
 
   constructor(
     private cd: ChangeDetectorRef,
-    private validationMessagesService: ValidationMessagesService
+    private validationMessagesService: ValidationMessagesService,
+    private controlContainer: ControlContainer,
+    @Host() protected host: MatFormField
```
Comment (mateuszbasinski) [inline projects/validation-messages/src/lib/components/validation-messages/validation-messages.component.ts:52]: protected dependencies should be before privates

File: projects/validation-messages/src/lib/components/validation-messages/validation-messages.component.ts [hunk 5]
```diff
@@ -85,6 +90,17 @@ export class ValidationMessagesComponent implements OnInit, OnDestroy, DoCheck {
     }
   }
 
+  get matInputControl() {
+    if (this.matInputRef && this.matInputRef.ngControl) {
```
Comment (mateuszbasinski) [inline projects/validation-messages/src/lib/components/validation-messages/validation-messages.component.ts:94]: i think `this.matInputRef?.ngControl` should works as well

File: projects/validation-messages/README.md (hunk 3 above)
Comment (dawidkostka-va) [reply projects/validation-messages/README.md]: fixed typo

File: projects/validation-messages/src/lib/components/validation-messages/validation-messages.component.ts [hunk 6]
```diff
@@ -85,6 +90,17 @@ export class ValidationMessagesComponent implements OnInit, OnDestroy, DoCheck {
     }
   }
 
+  get matInputControl() {
```
Comment (mateuszbasinski) [inline projects/validation-messages/src/lib/components/validation-messages/validation-messages.component.ts:93]: add type

File: projects/validation-messages/src/lib/components/validation-messages/validation-messages.component.ts [hunk 7]
```diff
@@ -182,4 +200,25 @@ export class ValidationMessagesComponent implements OnInit, OnDestroy, DoCheck {
...
+
+  private readFormControlByControlName() {
+    if (this.controlName === undefined) {
+      return;
+    }
+    const control = this.controlContainer.control?.get(this.controlName);
+    if (!(control instanceof FormControl)) {
+      return;
+    }
+    this.control = control as FormControl;
```
Comment (mateuszbasinski) [inline projects/validation-messages/src/lib/components/validation-messages/validation-messages.component.ts:212]:
```suggestion
    if (this.controlName === undefined) {
      return;
    }
    
    const control = this.controlContainer.control?.get(this.controlName);
    
    if (!(control instanceof FormControl)) {
      return;
    }
    
    this.control = control as FormControl;
```

File: projects/validation-messages/src/lib/components/validation-messages/validation-messages.component.ts [hunk 8]
```diff
@@ -182,4 +200,25 @@ export class ValidationMessagesComponent implements OnInit, OnDestroy, DoCheck {
...
+
+  private readFormControlFromHost() {
+    if (this.control !== undefined) {
+      return;
+    }
+    if (!(this.matInputControl instanceof FormControl)) {
+      return;
+    }
+    this.control = this.matInputControl;
+  }
```
Comment (mateuszbasinski) [inline projects/validation-messages/src/lib/components/validation-messages/validation-messages.component.ts:223]:
```suggestion
  private readFormControlFromHost() {
    if (this.control !== undefined) {
      return;
    }
    
    if (!(this.matInputControl instanceof FormControl)) {
      return;
    }
    
    this.control = this.matInputControl;
  }
```

File: projects/validation-messages/src/lib/components/validation-messages/validation-messages.component.ts [hunk 9]
```diff
@@ -182,4 +200,25 @@ export class ValidationMessagesComponent implements OnInit, OnDestroy, DoCheck {
     this.showServerErrors = false;
     this.valueChanges = null;
   }
+
+  private readFormControlByControlName() {
```
Comment (mateuszbasinski) [inline projects/validation-messages/src/lib/components/validation-messages/validation-messages.component.ts:204]: add type

File: projects/validation-messages/src/lib/components/validation-messages/validation-messages.component.ts [hunk 10]
```diff
@@ -182,4 +200,25 @@ export class ValidationMessagesComponent implements OnInit, OnDestroy, DoCheck {
...
+      return;
+    }
+    const control = this.controlContainer.control?.get(this.controlName);
+    if (!(control instanceof FormControl)) {
+      return;
+    }
+    this.control = control as FormControl;
+  }
+
+  private readFormControlFromHost() {
```
Comment (mateuszbasinski) [inline projects/validation-messages/src/lib/components/validation-messages/validation-messages.component.ts:215]: type

File: projects/validation-messages/src/lib/components/validation-messages/validation-messages.component.ts (hunk 7 above)
Comment (dawidkostka-va) [reply projects/validation-messages/src/lib/components/validation-messages/validation-messages.component.ts:212]: added whitespace

**Review by mateuszbasinski — APPROVED** [review]

**Pull request merged**